### DISCUSSION TITLE
How to get the original album, not the movie soundtrack song later appeared on

### DISCUSSION THREAD

**Discussion by rajb245** [shazamio/ShazamIO#50] [Q&A]
Here is a snippet:

```python
from shazamio import Shazam
import asyncio
import json

async def get_track_info(track_id):
    shazam = Shazam()
    about_track = await shazam.track_about(track_id=track_id)
    return about_track

track_id = 61541960 #https://www.shazam.com/track/61541960
loop = asyncio.get_event_loop()
shazam_metadata = loop.run_until_complete(get_track_info(track_id))
for section in shazam_metadata['sections']:
    if section['type'] == 'SONG': print(json.dumps(section['metadata'], indent=2))

print(shazam_metadata['images']['coverart'])
```

That will print out the album this song appears on and a URL to the coverart for the album. Both of those indicate "Safe Haven (Original Motion Picture Soundtrack)" is the album. As far as I can tell, the information of other albums does not appear anywhere in the response metadata I got back. However, if we go straight to the shazam site for the same track id (https://www.shazam.com/track/61541960), there is no mention of that movie soundtrack on the web UI and the album cover image shown there is for "Bear Creek", the actual album this song appeared on before the movie soundtrack.

So my main question here is: how do I get shazamio to give me the metadata for the album version, not this soundtrack? If this track ID has the right cover art and album on the web interface, why does the track_about() method return something different and how can I get the other metadata, the one I consider "correct"? There is a path I see using BeautifulSoup to scrape the website and not using shazamio at all, but this project probably came about because someone didn't want to write a scraper and properly reversed the API. Any suggestions?

**Comment by dotX12** [comment]
@rajb245,

https://www.shazam.com/discovery/v5/en-US/GB/web/-/track/61541960?shazamapiversion=v3&video=v3

GB:
![image](https://user-images.githubusercontent.com/64792903/219576638-56e18568-606e-43d4-b39f-a859021d46c4.png)



https://www.shazam.com/discovery/v5/en-US/RU/web/-/track/61541960?shazamapiversion=v3&video=v
![image](https://user-images.githubusercontent.com/64792903/219576701-e8eeb41b-8aaf-40f5-ab7d-1f896e5a8ae3.png)


Surprisingly, depending on the country code of the world, we see completely different results !!!

**Comment by dotX12** [comment]
I can add the ability to select your country code when create request if you need it.

**Reply by rajb245** [reply]
Thank you for the offer to update the interface of this library to query using different country codes. That is a generally useful feature. That would let me try different country codes to see if the song appears on different albums in different locales, but it still would not guarantee I'd get all the albums the song appears on, which would be needed to determine the first release of the song.

You can hold off modifying anything for now. After shazamio `recognize_song` gives me the basic info, I am looking at querying the iTunes Store and/or Apple Music APIs to see if they will return a canonical result for the album.

**Reply by dotX12** [reply]
Let me know if you need to make changes to the code later.

**Reply by rajb245** [reply]
Okay, so having shazamio be able to set the location for API calls set instead of being hardcoded to 'GB' seems to be useful. I was able to get iTunes IDs from shazamio (look for the 'id' key associated with value 'applemusicplay'), and indeed I am able to query for these in the [normal iTunes API](https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/iTuneSearchAPI/index.html) and get results. However, because the shazam endpoint is for Britain (GB), the iTunes IDs returned are specific to the GB iTunes store, and when I query the iTunes API, it sends me to the API endpoint based on my location (US), and sometimes, the US store IDs are not the same as the GB ones! Here's an example
https://music.apple.com/us/album/cool-kids/924487443?i=924487455
https://music.apple.com/gb/album/cool-kids/929409896?i=929409998
Same album, same song, but the ids are different, and the GB shazam returned the same one that is in the GB iTunes API.

I looked at how the URLs for the endpoints are defined in this codebase and it doesn't seem too hard to update it to accept an arbitrary country code. Just another parameter to `.format`, I think, and keeping the default 'GB' so the default behavior does not change. I could take a stab and submitting an MR, or I'd be happy if you could update the package. Thanks!

**Reply by rajb245** [reply]
Okay, I made PR #51 as a try at doing this myself; I'm a believer in collaboration, and I am open to feedback and updating the request to get it merged. Thanks!
